### PR TITLE
Change XML library for utilities/bcl

### PR DESCRIPTION
**Pull request by jasondegraw**
Replace the use of Qt's XML library with pugixml. This PR adds the pugixml library and changes the code for utilities/bcl.

### CODE REVIEW

**Comment by jasondegraw** [comment]
@macumber The commits so far get most of the use Qt, there's still some left in BCL.hpp and BCL.cpp (and maybe elsewhere, haven't made an exhaustive search yet). Before we merge this PR, I'll get that changed too, but I wanted to get the review ball rolling ASAP. Please take a look.

File: openstudiocore/src/pugixml/docs/samples/weekly-utf-8.xml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+<?xml version="1.0"?>
+<!DOCTYPE 週報 SYSTEM "weekly-utf-8.dtd">
+<!-- 週報サンプル -->
+<週報>
```
Comment (macumber) [inline openstudiocore/src/pugixml/docs/samples/weekly-utf-8.xml]: Hurray, UTF-8 tests!

**Comment by macumber** [comment]
@jasondegraw assuming the BCLXML tests pass after these changes this looks good to me, seems pretty straightforward

**Comment by macumber** [comment]
We will want to get rid of this cmake warning:

```
CMake Warning (dev) at src/pugixml/CMakeLists.txt:3 (project):
  Policy CMP0048 is not set: project() command manages VERSION variables.
  Run "cmake --help-policy CMP0048" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The following variable(s) would be set to empty:

    PROJECT_VERSION
    PROJECT_VERSION_MAJOR
    PROJECT_VERSION_MINOR
    PROJECT_VERSION_PATCH
This warning is for project developers.  Use -Wno-dev to suppress it.
```

and also all of these top level cmake options that get added (we can just hard code set them instead of making them options)

image

File: openstudiocore/CMakeLists.txt [hunk 2]
```diff
@@ -1366,6 +1371,9 @@ include_directories("${CMAKE_HOME_DIRECTORY}/src/")
 # For QWinWidget
 include_directories("${CMAKE_HOME_DIRECTORY}/src/qtwinmigrate")
 
+# XML
+include_directories("${CMAKE_HOME_DIRECTORY}/src/pugixml/src")
```
Comment (macumber) [inline openstudiocore/CMakeLists.txt]: This might be what triggered the huge rebuild

Comment (macumber) [reply openstudiocore/CMakeLists.txt]: I also added this which will eventually help, https://github.com/NREL/OpenStudio/pull/3373

File: openstudiocore/src/pugixml/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+project(pugixml)
+
+option(BUILD_SHARED_LIBS "Build shared instead of static library" OFF)
+option(BUILD_TESTS "Build tests" OFF)
+option(BUILD_PKGCONFIG "Build in PKGCONFIG mode" OFF)
+
+set(BUILD_DEFINES "" CACHE STRING "Build defines")
+
+if(MSVC)
+	option(STATIC_CRT "Use static CRT libraries" OFF)
```
[the remark this answers is not in the document]
Comment (jasondegraw) [reply openstudiocore/src/pugixml/CMakeLists.txt]: @macumber I'll defer to you on all CMake details

**Comment by jasondegraw** [comment]
@macumber On my local machine, I get unit test failures due to some networking things. So I'll need some assistance with figuring out what is going on, but it doesn't look like something I am causing.

**Comment by macumber** [comment]
Yes you can ignore the networking errors.  The only relevant error I saw was in BCLMeasure.BCLMeasure, something about an XML Checksum.  Seems like an error that can wait until later, no need to hold up the rest of the replacement

**Pull request merged**